mycore-mets/src/main/java/org/mycore/mets/tools/MCRMetsSave.java

```diff
@@ -949,4 +776,211 @@ public static String getFileId(MCRPath path) {
+                removePhysicalSubDivs(physicalSubDivsToRemove, divContainer);
+            }
+        }
+
+        private List<PhysicalSubDiv> removeFptrs(FileGrp fileGrp, PhysicalDiv divContainer, File fileToRemove) {
+            List<PhysicalSubDiv> physicalSubDivsToRemove = new ArrayList<>();
+            // remove file from mets:mets/mets:structMap[@TYPE='PHYSICAL']
+            for (PhysicalSubDiv physicalSubDiv : divContainer.getChildren()) {
+                List<Fptr> fptrsToRemove = new ArrayList<>();
+                for (Fptr fptr : physicalSubDiv.getChildren()) {
+                    if (fptr.getFileId().equals(fileToRemove.getId())) {
+                        if (fileGrp.getUse().equals(FileGrp.USE_MASTER)) {
+                            physicalSubDivsToRemove.add(physicalSubDiv);
+                        } else {
+                            fptrsToRemove.add(fptr);
+                        }
+                    }
+                }
+                for (Fptr fptrToRemove : fptrsToRemove) {
+                    LOGGER.info(() -> String.format(Locale.ROOT, "remove fptr \"%s\" from mets.xml of \"%s\"",
+                        fptrToRemove.getFileId(), file.getOwner()));
+                    physicalSubDiv.remove(fptrToRemove);
+                }
+            }
+            return physicalSubDivsToRemove;
+        }
+
+        private void removePhysicalSubDivs(List<PhysicalSubDiv> physicalSubDivsToRemove, PhysicalDiv divContainer) {
+            for (PhysicalSubDiv physicalSubDivToRemove : physicalSubDivsToRemove) {
+                //remove links in mets:structLink section
+                List<SmLink> list = mets.getStructLink().getSmLinkByTo(physicalSubDivToRemove.getId());
+                LogicalStructMap logicalStructMap = (LogicalStructMap) mets
+                    .getStructMap(LogicalStructMap.TYPE);
+
+                for (SmLink linkToRemove : list) {
+                    rmSmLink(linkToRemove, logicalStructMap);
+                }
+                divContainer.remove(physicalSubDivToRemove);
+            }
+        }
+
+        private void rmSmLink(SmLink linkToRemove, LogicalStructMap logicalStructMap) {
+            LOGGER.info(() -> String.format(Locale.ROOT, "remove smLink from \"%s\" to \"%s\"",
+                linkToRemove.getFrom(), linkToRemove.getTo()));
+            mets.getStructLink().removeSmLink(linkToRemove);
+            // modify logical struct Map
+            String logID = linkToRemove.getFrom();
+            // the deleted file was not directly assigned to a structure
+            if (logicalStructMap.getDivContainer().getId().equals(logID)) {
+                return;
+            }
+            LogicalDiv logicalDiv = logicalStructMap.getDivContainer().getLogicalSubDiv(logID);
+            if (logicalDiv == null) {
+                LOGGER.error("Could not find 'LogicalDiv' with id '{}'.", () -> logID);
+                return;
```

Comment:
```
if (logicalDiv == null) {
    LOGGER.error("Could not find {} with id {}",
        () -> LogicalDiv.class.getSimpleName(), () -> logID);
    LOGGER.error("Mets document remains unchanged");
    return mets;
}
```
I'm not that familiar with Mets, maybe it's an optimization, but the behavior doesn't seem to be equivalent.